sympy/functions/elementary/trigonometric.py

```diff
@@ -468,7 +468,10 @@ def _eval_as_leading_term(self, x):
         if x in arg.free_symbols and Order(1, x).contains(arg):
             return arg
         else:
-            return self.func(arg)
+            if not self.is_meromorphic(x, 0):
```

Comment:
This is true when `arg.subs(x, 0)` is not finite. That could be tested directly.